Coral/Coral.js

```diff
@@ -0,0 +1,312 @@
+  }
+}
+
+const SPLINE = new Spline([
+  new ControlPoint(0.75, 0.0, -0.21, 0.219),
+  new ControlPoint(0.75, 0.5, 0.09, 0.204),
+  new ControlPoint(0.5, 0.934, -0.185, 0.055),
+  new ControlPoint(0.25, 0.5, 0.11, -0.291),
+  new ControlPoint(0.25, 0.0, -0.095, -0.176),
+]);
+
+const CONSTRAINTS = [
+  new Rect(0.5, 0.0, 0.5, 0.0),
+  new Rect(0.5, 0.5, 0.5, 0.0),
+  new Rect(0.5, 0.5, 0.0, 0.5),
+  new Rect(0.0, 0.5, 0.5, 0.0),
+  new Rect(0.0, 0.0, 0.5, 0.0),
+];
+
+const VERTICES = SPLINE.control_points.map(
+  (x, i) => new InteractiveVertex(x, CONSTRAINTS[i])
+);
+
+const TANGENT_CONSTRAINTS = [
+  new Rect(-0.5, 0, 1.0, 0.5),
+  new Rect(-0.5, 0, 1.0, 0.5),
+  new Rect(-0.5, -0.5, 0.5, 1.0),
+  new Rect(-0.5, -0.5, 1.0, 0.5),
+  new Rect(-0.5, -0.5, 1.0, 0.5),
+];
+
+const TANGENTS = SPLINE.control_points.map(
+  (x, i) => new InteractiveTangent(x, TANGENT_CONSTRAINTS[i])
+);
+
+export const sketch = (p) => {
+  let canvas;
+  p.setup = () => {
+    canvas = p.createCanvas(500, 700).elt;
+  };
+
+  p.draw = () => {
+    p.background(0);
+
+    p.stroke(127);
+    p.noFill();
+    p.rect(QUAD_X, QUAD_Y, QUAD_WIDTH, QUAD_HEIGHT);
+    p.push();
+    p.translate(p.width / 2, p.height / 2);
+    p.line(0, -100, 0, 100);
+    p.line(-100, 0, 100, 0);
+    p.pop();
+
+    // Draw the Bezier spline
+    p.stroke(131, 71, 181);
+    p.noFill();
+    p.strokeWeight(2);
+    for (let i = 0; i < SPLINE.control_points.length - 1; i++) {
+      const start = SPLINE.control_points[i];
+      const end = SPLINE.control_points[i + 1];
+
+      const a = uv_to_world(start.position);
+      const b = uv_to_world(start.forward_point);
+      const c = uv_to_world(end.backward_point);
+      const d = uv_to_world(end.position);
+
+      p.bezier(a.x, a.y, b.x, b.y, c.x, c.y, d.x, d.y);
+    }
+
+    // Draw tangents
+    p.stroke(0, 255, 0);
+    p.noFill();
+    p.strokeWeight(1);
+    for (const point of SPLINE.control_points) {
+      const a = uv_to_world(point.position);
+      const b = uv_to_world(point.forward_point);
+      p.line(a.x, a.y, b.x, b.y);
+    }
+    // Draw a circle at the tips of the tangents
+    p.stroke(0, 255, 0);
+    p.noFill();
+    p.strokeWeight(6);
+    for (const point of SPLINE.control_points) {
+      const pos = uv_to_world(point.forward_point);
+      p.point(pos.x, pos.y);
+    }
+
+    // Draw vertices
+    p.stroke(255, 255, 0);
+    p.strokeWeight(6);
+    p.noFill();
+    for (const point of SPLINE.control_points) {
+      const pos = uv_to_world(point.position);
+      p.point(pos.x, pos.y);
+    }
+
+    p.stroke(255);
+    p.noFill();
+    p.strokeWeight(1);
+    if (selected_tangent) {
+      const position = uv_to_world(selected_tangent.tip);
+      p.circle(position.x, position.y, SELECT_RADIUS * 2);
+    } else if (selected_vertex) {
+      const position = uv_to_world(selected_vertex.position);
+      p.circle(position.x, position.y, SELECT_RADIUS * 2);
+    }
+  };
+
+  let mouse = { x: 0, y: 0 };
```

Comment:
Dead code, this can be removed. I think even some of the callbacks can be removed too.